### ISSUE TITLE
closest/farthest/average precision

### ISSUE THREAD

**Issue by kylekatarnls** [briannesbitt/Carbon#1512]
Results of closest, farthest and average are wrong when difference is less than 1 second:
```php
$baseDate = Carbon::parse('2018-10-11 20:59:06.500000');
$closestDate = Carbon::parse('2018-10-11 20:59:06.600000');
$farthestDate = Carbon::parse('2018-10-11 20:59:06.300000');

echo $baseDate->closest($closestDate, $farthestDate)->format('s.u');
```
Carbon version: **2.5.2**
PHP version: **7.2.12**

Expected: `06.600000`
Actually: `06.300000`

**Comment by kylekatarnls** [comment]
Fix for Carbon 2: #1520

**Comment by kylekatarnls** [comment]
Fix for Carbon 1: #1521

**Comment by kylekatarnls** [comment]
Fixed, and ready for next releases
